src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertyAndValueDeserializer.cs

```diff
@@ -728,6 +703,67 @@ in resourceState.PropertyAndAnnotationCollector.GetODataPropertyAnnotations(nest
             return ODataJsonLightReaderNestedResourceInfo.CreateResourceSetReaderNestedResourceInfo(nestedResourceInfo, navigationProperty, propertyType, expandedResourceSet);
         }
 
+/*
+        /// <summary>
+        /// Reads a nested stream
+        /// </summary>
+        /// <param name="resourceState">The state of the reader for resource to read.</param>
+        /// <param name="property">The property for which to read the nested resource info. null for undeclared property.</param>
+        /// <param name="propertyName">The property name.</param>
+        /// <returns>The nested resource info for the stream collection.</returns>
+        /// <remarks>
+        /// This method doesn't move the reader.
+        /// </remarks>
+        protected static ODataJsonLightReaderNestedInfo ReadStreamNestedResourceInfo(IODataJsonLightReaderResourceState resourceState, IEdmStructuralProperty property, string propertyName)
+        {
+            Debug.Assert(resourceState != null, "resourceState != null");
+            Debug.Assert((propertyName != null) || (property != null),
+                "The collection property and property name shouldn't both be null");
+            Debug.Assert(property == null || property.Type.Definition.IsStream(),
+                "The type of the property, if specified, must be Edm.Stream");
+
+            ODataNestedResourceInfo nestedResourceInfo = new ODataNestedResourceInfo()
+            {
+                Name = propertyName,
+                IsCollection = false,
+                IsComplex = false
+            };
+
+            return ODataJsonLightReaderNestedResourceInfo.CreateResourceReaderNestedResourceInfo(nestedResourceInfo, property, EdmCoreModel.Instance.GetPrimitiveType(EdmPrimitiveTypeKind.Stream));
+        }
+*/
+
+        /// <summary>
+        /// Reads a nested stream collection as a resource set.
+        /// </summary>
+        /// <param name="resourceState">The state of the reader for resource to read.</param>
+        /// <param name="collectionProperty">The collection of stream property for which to read the nested resource info. null for undeclared property.</param>
+        /// <param name="propertyName">The property name.</param>
+        /// <param name="elementType">They primitive type of the collection element</param>
+        /// <returns>The nested resource info for the stream collection.</returns>
+        /// <remarks>
+        /// This method doesn't move the reader.
+        /// </remarks>
+        protected static ODataJsonLightReaderNestedResourceInfo ReadStreamCollectionNestedResourceInfo(IODataJsonLightReaderResourceState resourceState, IEdmStructuralProperty collectionProperty, string propertyName, IEdmType elementType)

```

Comment:
>ODataJsonLightReaderNestedResourceInfo [](start = 25, length = 38)

This method should return "ODataJsonLightReaderNestedStreamInfo", Right?

Reply:
No, actually it's the nestedStreamInfo for the resource set that happens to contain stream values, not for the stream values.

---
In reply to: [259083359](https://github.com/OData/odata.net/pull/1204#discussion_r259083359) [](ancestors = 259083359)